src/coreclr/nativeaot/Runtime/Portable/CMakeLists.txt

```diff
@@ -41,3 +41,10 @@ if (NOT CLR_CMAKE_TARGET_ARCH_WASM)
   install_static_library(standalonegc-enabled aotsdk nativeaot)
 endif()
 
+if (CLR_CMAKE_TARGET_WASI)
```

Comment:
I'm not sure but should this be part of `libSystem.Native.a` rather than part of NAOT tree ?
So that we could use it in Mono too.

I'll try to do it in my PR and see how it goes.

Reply:
I've been deliberately avoiding changes that might affect Mono, but only because I wasn't sure how to test them.  But given what you're doing, we should definitely share code as appropriate.

Reply:
adding the `.o` or  `.wit ` into `.a` is not pulled by `wasm-component-ld`.
Also with `-Wl,--component-type,WasiHttpWorld_component_type.wit` I need to make it separate parameter

Reply:
I guess this should be just another 'native' artifact type that is installed in cmake by itself (similar to Emscripten JS libraries in that way).

Reply:
yeah, but it's mainly driven by msbuild for mono wasi workload.

Reply:
Here's how I imagined it would work:

1) There is a `.wit` in the source tree of `src/native`. It is installed by cmake into the same place where all the other native artifacts are installed.
2) It is picked up by the usual runtime pack formation logic and placed alongside those other native artifacts.
3) There is linking logic that adds them to the final link.
   - For NAOT this logic will live `Microsoft.NETCore.Native.targets`.
   - For Mono, I imagine it'll need to live both in the runtime build project that produces the precompiled runtime, and somewhere else for on-user-machine linking.

So basically exactly the same logic as for `libSystem.*.a`, only difference is the linker argument name.